src/options.ts

```diff
@@ -87,7 +89,11 @@ export function _setDefaultOptions(options: Partial<Options> = {}): Options {
 
   const extraTracerConfig = options.tracerConfig || {};
   const tracerConfig = {
-    resource: new EnvResourceDetector().detect(),
+    resource: new EnvResourceDetector().detect().merge(
+      new Resource({
+        [ResourceAttributes.SERVICE_NAME]: options.serviceName,
```

Comment:
If service name is not provided via options but is provided with `OTEL_RESOURCE_ATTRIBUTES` or `OTEL_SERVICE_NAME` in future, will it get ovewritten by the value of `defaultServiceName` here?

Reply:
The precedence is now `options.serviceName > env.SPLUNK_SERVICE_NAME > service.name from env.OTEL_RESOURCE_ATTRIBUTES > defaultServiceName`